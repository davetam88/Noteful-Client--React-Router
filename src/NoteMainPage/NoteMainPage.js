import React, { Component } from 'react';
import Note from '../Note/Note'
import NotefulContext from '../App/NotefulContext';
import { findNote } from '../notes-helpers'
import './NoteMainPage.css'
import PropTypes from 'prop-types';

export default class NoteMainPage extends Component {

  static defaultProps = {
    match: {
      params: {}
    }
  }

  static contextType = NotefulContext

  handleDeleteNote = noteId => {
    this.props.history.push(`/`)
  }

  render() {

    const { notes = [] } = this.context
    const { noteId } = this.props.match.params
    const xnoteId = Number(noteId);
    const note = findNote(notes, xnoteId) || { content: '' }

    return (
      <section className='NoteMainPage' >

        <Note
          id={note.id}
          name={note.name}
          modified={note.modified}
          onDeleteNote={this.handleDeleteNote}
        />

        <div className='NoteMainPage__content'>
          {note.content.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>{para}</p>
          )}
        </div>
      </section>
    )
  }
}

NoteMainPage.propTypes = {
  params: PropTypes.object
}
