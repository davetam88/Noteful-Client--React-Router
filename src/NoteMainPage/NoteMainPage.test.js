import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteMainPage from './NoteMainPage'

describe(`NoteMainPage component`, () => {
  const props = {
    note: {
      "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Dogs",
      "modified": "2019-01-03T00:00:00.000Z",
      // "folderId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
      "content": "Corporis accusamus placeat.\n \rUnde."
    }
  }

  it('renders a .NoteMainPage by default', () => {
    const wrapper = shallow(<NoteMainPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Note with note prop', () => {
    const note = shallow(<NoteMainPage {...props} />)
      .find('Note')
    expect(toJson(note)).toMatchSnapshot()
  })

  it(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    [{
      note: { "content": "Content with n r.\n \rafter n r." }
    }, {
      note: { "content": "Content with n.\nafter." }
    }].forEach(props => {
      const content = shallow(<NoteMainPage {...props} />)
        .find('NoteMainPage__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})
