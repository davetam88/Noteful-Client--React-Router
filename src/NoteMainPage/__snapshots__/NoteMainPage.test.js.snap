// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteMainPage component renders a .NoteMainPage by default 1`] = `
<section
  className="NoteMainPage"
>
  <Note
    onDeleteNote={[Function]}
  />
  <div
    className="NoteMainPage__content"
  >
    <p
      key="0"
    />
  </div>
</section>
`;

exports[`NoteMainPage component renders a Note with note prop 1`] = `
<Note
  onDeleteNote={[Function]}
/>
`;

exports[`NoteMainPage component splits the content by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`NoteMainPage component splits the content by \\n or \\n\\r, with a p foreach 2`] = `null`;
